docs/user/implement_env.md

```diff
@@ -0,0 +1,312 @@
+
+The attributes of :class:`Environment` are:
+
+| Properties                              | Description                 |
+| --------------------------------------- | --------------------------- |
+| :meth:`~Environment.action_space`       | The action space specification.      |
+| :meth:`~Environment.observation_space`  | The observation space specification. |
+| :meth:`~Environment.spec`               | The environment specifications.      |
+| :meth:`~Environment.render_modes`       | The list of supported render modes.  |
+
+You should expect to use all environments in garage with this API.
+
+## Add Support for an Existing Environment Library
+
+Luckily, Garage already came with a variety of external environment wrappers:
+
+* :class:`GymEnv`, which supports the `gym` API.
+* :class:`DmControlEnv`, which supports Deepmind's
+`dm_control` API.
+* :class:`BulletEnv`, which supports the `pybullet` API.
+
+    Note that `pybullet` environments are `gym`-based environments, and they
+    can be wrapped with :class:`GymEnv` as well. In such case,
+    :class:`GymEnv` will detect the bullet-based environment and returns
+    a :class:`BulletEnv`.
+
+If you would like to add an existing environment to garage, you will probably
+need an environment wrapper to handle environment-specific logic.
+
+Here is an handy To-do list for adding a new environment wrapper:
+
+* (Recommended) **Inherit** the environment wrapper from :class:`Environment`.
+
+    Note that this is not required in Python because of duck typing. Your
+    custom environment class will work as long as it implements
+    interface methods of :class:`Environment`.
+
+* **Implement** attributes: `observation_space`, `action_space`, `spec`,
+and `render_modes`
+
+    Note that `observation_space` and `action_space` need to be `akro.Space`.
+
+* **Implement** public methods: `reset()`, `step()`, `render()`, `visualize()`,
+`close()`
+
+* (Optional) **Implement** `__getstate__` and `__setstate__`
+
+    Garage pickles the environment to save snapshots, whereas some attributes
+    of your environment might not be pickle-able (e.g. a client-server
+    connection). If needed, provide your own implementation to make sure your
+    environment is pickle-able.
+
+* (Optional) **Unit Tests**
+
+    If you would like to contribute your own environment back to garage,
+    make sure to add unit tests under the directory `tests/garage/envs`.
+    You can find examples of environment tests here as well.
+
+
+## Implement a New Environment
+
+In the rest of this section, we will walk through an example of implementing a
+point robot environment using our framework. A more complete version of this
+environment is available as :class:`PointEnv`.
+
+We will implement a simple environment with 2D observations and 2D actions. The
+goal is to control a point robot in 2D to move it to the origin. We receive
+position of a point robot in the 2D plane :math:`(x, y) \in \mathbb{R}^2`.
```

Comment:
These `:math:` don't seem to work either.

Reply:
fixed with `eval_rst`